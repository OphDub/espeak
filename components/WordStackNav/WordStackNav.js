import React, { Component } from 'react';
import { StackNavigator } from 'react-navigation';
import Decks from '../Decks/Decks';
import WordCards from '../WordCards/WordCards';

const routeConfig = {
  Decks: {
    screen: Decks
  },
  WordCards: {
    screen: WordCards
  }
};

const RootNav = StackNavigator(routeConfig);

type Props = {};
export default class WordStackNav extends Component<Props> {
  render() {
    const {
      handlePoints,
      handleHintPoints,
      userPoints,
      user,
      decks,
      updateUser
    } = this.props.screenProps;

    return (
      <RootNav screenProps={{
        handlePoints,
        handleHintPoints,
        userPoints,
        user,
        decks,
        updateUser
      }}/>
    );
  }
}