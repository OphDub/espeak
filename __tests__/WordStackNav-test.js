import React from 'react';
import { configure } from 'enzyme';
import WordStackNav from '../components/WordStackNav/WordStackNav';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import decks from '../mockData/decks';
configure({ adapter: new Adapter() });

describe('WORD STACK NAV', () => {
  it('renders correctly', () => {
    const user = {
      username: 'opa',
      email: 'opa@startup.com',
      points: 0,
      stack_id: 1,
      firebase_id: 12345
    };
    const mockProps = {
      handlePoints: jest.fn(),
      user,
      updateUserPoints: jest.fn(),
      decks,
    }
    const wrap = renderer.create(<WordStackNav screenProps={mockProps} />).toJSON();
    expect(wrap).toMatchSnapshot();
  });
});