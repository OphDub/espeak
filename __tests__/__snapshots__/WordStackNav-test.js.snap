// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WORD STACK NAV renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#EFEFF4",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "paddingVertical": 40,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#B7B2B8",
                  "borderRadius": 8,
                  "margin": 10,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                Basics 1
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#B7B2B8",
                  "borderRadius": 8,
                  "margin": 10,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                Basics 2
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#B7B2B8",
                  "borderRadius": 8,
                  "margin": 10,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                Food 1
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#B7B2B8",
                  "borderRadius": 8,
                  "margin": 10,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                Food 2
              </Text>
            </View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#B7B2B8",
                  "borderRadius": 8,
                  "margin": 10,
                  "opacity": 1,
                  "padding": 5,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
                }
              >
                Verbs
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "borderBottomColor": "#A7A7AA",
          "borderBottomWidth": 0.5,
          "height": 64,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "flexDirection": "row",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={undefined}
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "bottom": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessibilityTraits="header"
              accessible={true}
              allowFontScaling={true}
              collapsable={undefined}
              ellipsizeMode="tail"
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Object {
                  "color": "rgba(0, 0, 0, .9)",
                  "fontSize": 17,
                  "fontWeight": "700",
                  "marginHorizontal": 16,
                  "textAlign": "center",
                }
              }
            >
              Decks
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
